omnigibson/envs/env_base.py

```diff
@@ -592,6 +604,7 @@ def step(self, action):
             action (gym.spaces.Dict or dict or np.array): robot actions. If a dict is specified, each entry should
                 map robot name to corresponding action. If a np.array, it should be the flattened, concatenated set
                 of actions
+            n_render_iterations (int): Number of rendering iterations to use before returning observations
```

Comment:
Why is this an argument here and not using the render frequency? What is the purpose of having an additional n_render_iterations here? I really think this doesnt belong in this API, it's way too low-level, plus we're now changing how many times a render will happen per step?

Reply:
cf. eric's comment above. It's a two-fold issue here:

1. Omni's real-time raytracing, for whatever reason, is both laggy and jittery. By laggy I mean that the actual rendered frame is always a few frames BEHIND what the actual physical state of the world is. So if you teleport the robot you need to render multiple times before you actually observe the change in the GUI / observation.
2. We need the rendered frame to be accurate _before_ we grab observations. This means that we can't simply do additional renders after env.step() gets called, because at the point the observations have already been recorded and they're not exactly accurate with respect to the most up-to-date physical state of the world (assuming we've only rendered once).

Reply:
resolved in person